src/coreclr/jit/optimizer.cpp

```diff
@@ -10247,3 +10247,14 @@ void Compiler::optMarkLoopRemoved(unsigned loopNum)
 // `fgDebugCheckLoopTable()` is called.
 #endif // DEBUG
 }
+
+ValueNum Compiler::optConservativeVN(GenTree* tree)
+{
+    if (optLocalAssertionProp)
```

Comment:
@dotnet/jit-contrib Is there a reasonable way to avoid this redundant check?

Reply:
What it the redundant part here?

Reply:
I think this is the most straightforward way to do that's also consistent with the rest of assertion prop.

We have both local and global assertion prop that share this code. The former runs before value numbering while the latter runs after and derives facts from value numbering. The reason the code works today is that we end up in `ValueNumStore::GetVNFunc` that immediately exits if the value number is `NoVN`, so either outlining this check or duplicating the check here (which would presumably allow the native compiler to remove one of them after inlining; these functions are pretty small) would remove the redundancy. However, TP wise it does not look significant and the rest of assertion prop is conditioning VN uses on this bool, so I think this is fine.